src/openms/include/OpenMS/FORMAT/DATAACCESS/SwathFileConsumer.h

```diff
@@ -631,7 +634,7 @@ namespace OpenMS
         addMS1Map_();
       }
       ms1_consumer_->consumeSpectrum(s);
-      s.clear(false);
+      ms1_map_->addSpectrum(s); // keep spectrum in memory
```

Comment:
you cannot do that, this will keep all MS1 spectra in memory. We have data with MS1 maps > 50 GB and that is not going to work for us. Why dont you use a consumer that keeps data in memory and retrieve the data from there? There should actually be one that does this the `MSDataStoringConsumer.cpp` it will do that for you and you dont have to change this interface here!